integration_tests/src/main/python/parquet_test.py

```diff
@@ -1145,3 +1145,23 @@ def test_parquet_read_encryption(spark_tmp_path, reader_confs, v1_enabled_list):
         lambda: with_gpu_session(
             lambda spark: spark.read.parquet(data_path).collect(), conf=conf),
         error_message='The GPU does not support reading encrypted Parquet files')
+
+def test_parquet_read_count(spark_tmp_path):
+    parquet_gens = [byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen,
+    string_gen, boolean_gen, DateGen(start=date(1590, 1, 1)),
+    TimestampGen(start=datetime(1900, 1, 1, tzinfo=timezone.utc))] + decimal_gens
+    gen_list = [('_c' + str(i), gen) for i, gen in enumerate(parquet_gens)]
+    data_path = spark_tmp_path + '/PARQUET_DATA'
+
+    with_cpu_session(lambda spark: gen_df(spark, gen_list).write.parquet(data_path))
+
+    assert_gpu_and_cpu_row_counts_equal(lambda spark: spark.read.parquet(data_path))
+
+    def do_parquet_read(spark):
+        df = spark.read.parquet(data_path)
+        df.createOrReplaceTempView("tab")
+        return spark.sql("SELECT COUNT(*) FROM tab")
+
+    # assert the spark plan of the equivalent SQL query contains no column in read schema
+    assert_cpu_and_gpu_are_equal_collect_with_capture(
+        do_parquet_read, exist_classes="ReadSchema: struct<>")
```

Comment:
Can we add the `exec` operator, like:
`exist_classes="GpuFileGpuScan parquet .*  ReadSchema: struct<>"`